File: Tree/Tree Traversal/Inorder Traversal (Recursive)/sol.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def inorderTraversal(self, root: TreeNode) -> List[int]:
        res = []
        if root:
            res += self.inorderTraversal(root.left) + [root.val] + self.inorderTraversal(root.right)
        return res
    
    
    # inorder : L->Root->R
    # preorder : Root->L->R
    # postorder : L->R->Root
        

File: Tree/Tree Traversal/Inorder Traversal (Recursive)/sol.c
#include <stdio.h>
#include <malloc.h>


// declaring node of Binary-Tree
struct BinaryTreeNode {
    struct BinaryTreeNode *left;
    int info;
    struct BinaryTreeNode *right;
};


struct BinaryTreeNode *root=NULL;



// function for inOrder Traversal (Recursive Approach)
void inorderTraversal(struct BinaryTreeNode *root) {
    if(root == NULL) {
        return;
    }
    else {
        inorderTraversal(root->left);
        printf(" %d ", root->info);
        inorderTraversal(root->right);
    }
}
